sklearn/cluster/meson.build

```diff
@@ -5,20 +5,20 @@ cluster_extension_metadata = {
     {'sources': ['_hierarchical_fast.pyx', metrics_cython_tree],
      'override_options': ['cython_language=cpp']},
   '_k_means_common':
-    {'sources': ['_k_means_common.pyx']},
+    {'sources': ['_k_means_common.pyx'], 'dependencies': [openmp_dep]},
   '_k_means_lloyd':
-    {'sources': ['_k_means_lloyd.pyx']},
+    {'sources': ['_k_means_lloyd.pyx'], 'dependencies': [openmp_dep]},
   '_k_means_elkan':
-    {'sources': ['_k_means_elkan.pyx']},
+    {'sources': ['_k_means_elkan.pyx'], 'dependencies': [openmp_dep]},
   '_k_means_minibatch':
-    {'sources': ['_k_means_minibatch.pyx']},
+    {'sources': ['_k_means_minibatch.pyx'], 'dependencies': [openmp_dep]},
 }
 
 foreach ext_name, ext_dict : cluster_extension_metadata
   py.extension_module(
     ext_name,
     [ext_dict.get('sources'), utils_cython_tree],
-    dependencies: [np_dep, openmp_dep],
+    dependencies: [np_dep] + ext_dict.get('dependencies', []),
```

Comment:
does this now make things more specific since we have the linter?

Reply:
The linter will catch if an extension module use OpenMP flags but does not use it, where "using OpenMP" is defined by the following `git grep` regexp. I think it makes sense but please double-check:
```
git grep -lP "cython.*parallel|_openmp_helpers"
```